search-service/src/main/kotlin/com/egm/stellio/search/service/TemporalEntityAttributeService.kt

```diff
@@ -714,6 +768,78 @@ class TemporalEntityAttributeService(
         }
     }
 
+    @Transactional
+    suspend fun mergeEntityAttributes(
+        entityUri: URI,
+        ngsiLdAttributes: List<NgsiLdAttribute>,
+        expandedAttributes: ExpandedAttributes,
+        teas: List<TemporalEntityAttribute>,
+        createdAt: ZonedDateTime,
+        observedAt: ZonedDateTime?,
+        sub: Sub?
+    ): Either<APIException, UpdateResult> =
+        either {
+            val attributeInstances = ngsiLdAttributes.flatOnInstances()
+            attributeInstances.parTraverseEither { (ngsiLdAttribute, ngsiLdAttributeInstance) ->
+                logger.debug("Merging attribute {} in entity {}", ngsiLdAttribute.name, entityUri)
+                val currentTea = teas.firstOrNull {
+                    it.attributeName == ngsiLdAttribute.name && it.datasetId == ngsiLdAttributeInstance.datasetId
+                }
+                var attributeMetadata = ngsiLdAttributeInstance.toTemporalAttributeMetadata().bind()
+                var attributePayload = getAttributeFromExpandedAttributes(
+                    expandedAttributes,
+                    ngsiLdAttribute.name,
+                    ngsiLdAttributeInstance.datasetId
+                )!!
+
+                if (
+                    observedAt != null &&
+                    currentTea?.payload.toString().contains(NGSILD_OBSERVED_AT_PROPERTY) &&
+                    !attributePayload.containsKey(NGSILD_OBSERVED_AT_PROPERTY)
+                ) {
+                    attributePayload =
+                        attributePayload.plus(NGSILD_OBSERVED_AT_PROPERTY to buildNonReifiedDateTime(observedAt))
+                    attributeMetadata = attributeMetadata.copy(observedAt = observedAt)
+                }
```

Comment:
since it is specific to the merge of an existing attribute, maybe it could be moved there?